eo-maven-plugin/src/main/java/org/eolang/maven/rust_project/Names.java

```diff
@@ -0,0 +1,137 @@
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven.rust_project;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.concurrent.ConcurrentHashMap;
+import org.cactoos.list.ListOf;
+import org.eolang.maven.footprint.FtDefault;
+import org.eolang.maven.util.Home;
+
+/**
+ * Uniquely converts the insert into the name for jni function.
+ * @since 0.29
+ */
+public final class Names {
+
+    /**
+     * Target directory.
+     * @checkstyle MemberNameCheck (7 lines)
+     */
+    private final Path dest;
+
+    /**
+     * All names.
+     */
+    private final ConcurrentHashMap<String, String> all;
+
+    /**
+     * Prefix for the name.
+     */
+    private final String prefix;
+
+    /**
+     * Ctor.
+     * @param target Directory where to serialize names.
+     * @throws IOException If any issues with IO.
+     */
+    @SuppressWarnings("PMD.ConstructorOnlyInitializesOrCallOtherConstructors")
+    public Names(final Path target) throws IOException {
+        this.dest = target.resolve("names");
+        this.prefix = target.toString().toLowerCase(Locale.ENGLISH).replaceAll("[^a-z0-9]", "x");
+        if (this.dest.toFile().exists()) {
+            this.all = Names.load(this.dest);
+        } else {
+            this.all = new ConcurrentHashMap<>();
+        }
+    }
+
+    /**
+     * Assign the new name to the function.
+     * @param code Code of the function.
+     * @param dependencies Function dependencies.
+     * @return The name.
+     */
+    public String name(final String code, final String dependencies) {
+        return this.all.computeIfAbsent(
+            new ListOf<>(code, dependencies).toString(),
+            key -> String.format(
+                "%s%d",
+                this.prefix,
+                this.all.size()
+            )
+        );
+    }
+
+    /**
+     * Saves the function to name dispatching table.
+     * @throws IOException If any issues with IO.
+     */
+    public void save() throws IOException {
+        Files.createDirectories(this.dest.getParent());
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        final ObjectOutputStream oos = new ObjectOutputStream(baos);
```

Comment:
@levBagryansky Maybe we can save to a plain `txt` file instead? Something like CSV. In this case we will be able to see the content of the file. And the implementation will be much simpler I guess.

Reply:
@levBagryansky I am not sure, of course. We have to check it. I seems that `csv` just simpler to implement in the code and `csv` file could be read by a human.

Reply:
@volodya-lombrozo I have checked it in pr #2028. Serialization takes `~1200` and csv takes `~1700` on my machine.

Reply:
@levBagryansky It's not such a big penalty, as for me. Moreover, I  think that we can insert comma between two strings without additional library.
But ok, I don't insist here. If you want leave objects - let's do it. We always will have a chance to refactor the code if it will be needed.

Reply:
@volodya-lombrozo we cannot just insert comma between strings because `string`s can contain any symbol. That is why we would to think of right formatting.